### ISSUE TITLE
Networking v2: LBaaS "LoadBalancer" object uses tenant_id instead of project_id

### ISSUE THREAD

**Issue by waynr** [gophercloud/gophercloud#1149]
First off, I want to establish that this API uses the same URLs as a differently-pathed API in gophercloud that seems to correctly use `project_id`:

* https://github.com/gophercloud/gophercloud/blob/f1bae72a73954e0636d2de309b417bcfa4bd7de7/openstack/loadbalancer/v2/loadbalancers/urls.go#L6-L8
* https://github.com/gophercloud/gophercloud/blob/f1bae72a73954e0636d2de309b417bcfa4bd7de7/openstack/networking/v2/extensions/lbaas_v2/loadbalancers/urls.go#L6-L8

* https://github.com/gophercloud/gophercloud/blob/f1bae72a73954e0636d2de309b417bcfa4bd7de7/openstack/loadbalancer/v2/loadbalancers/results.go#L21
* https://github.com/gophercloud/gophercloud/blob/f1bae72a73954e0636d2de309b417bcfa4bd7de7/openstack/networking/v2/extensions/lbaas_v2/loadbalancers/results.go#L21

Oh I guess I already filed a similar ticket in the past where I referenced octavia itself: #964 

I guess in my original issue I neglected to notice that I was linking to the wrong file in gophercloud since my issue (always receiving an empty TenantID) seems to persist.

**Comment by waynr** [comment]
Also worth noting is that I can't simply use the other API path in gophercloud since it doesn't include the `CascadingDelete` function that my code is using.

**Comment by jtopjian** [comment]
@waynr To possibly clear up any confusion:

`CascadingDelete` is only part of Octavia (`gophercloud/openstack/loadbalancer/v2`). You should use a [`LoadBalancer`](https://github.com/gophercloud/gophercloud/blob/master/openstack/client.go#L383-L389) client for this.

Technically you can use both a networking client and load balancer client to interact with the `gophercloud/openstack/loadbalancer/v2` and `gophercloud/openstack/networking/v2/extensions/lbaas_v2` because there's such a big overlap between the two projects' API features. But Octavia should only be used in the former or else you'll run into errors trying to use features such as `CascadingDelete` when it's triggered by a networking client.

The Neutron/Networking LBaaS v2 API is still a legitimate API, but only for networking-based clients. I imagine in a year or two this could be considered old and removeable from Gophercloud.

On the topic of `tenant_id` / `project_id`, there's some background on this here: https://github.com/gophercloud/gophercloud/issues/790 but it's for the Network service in general. Still might be helpful.

**Comment by jrperritt** [comment]
> I don't see CascadingDelete in that package

Correct, we wrapped it into [`Delete`](https://github.com/gophercloud/gophercloud/blob/master/openstack/loadbalancer/v2/loadbalancers/requests.go#L176)

**Comment by jtopjian** [comment]
> Correct, we wrapped it into Delete

Yes, that's my mistake. I forgot we did it this way.

**Comment by waynr** [comment]
Ah, nice thanks for pointing that out! In that case this issue isn't particularly urgent for me.
